### PR TITLE
Enable Secure, HttpOnly, Path and Expires for some cookies in PR Status

### LINKED ISSUE #7233: PR status github oauth token needs to be marked secure and http
Even though this is only capable of doing public queries, it should only be sent over HTTPs and shouldn't be accessible to JS. The username token should also be set to "secure" but not HTTP only.

### LINKED ISSUE #7234: PR status page doesn't display username in the first-load query
/kind bug
/area prow

The first time you load the PR page (delete all cookies first), sign in, and then go back to the PR page you will see `is:pr state:open author:`. The query results are still `author:you` but we don't redirect to this URL correctly.

### PR DESCRIPTION
**Pull request by qhuynh96**
github_login cookie's path should be set to '/' so that the client side can access it.
* Fixes #7234
* Fixes #7233

### CODE REVIEW

File: prow/githuboauth/githuboauth.go [hunk 1]
```diff
@@ -160,14 +161,17 @@ func (ga *GithubOAuthAgent) HandleRedirect(client OAuthClient, getter GithubClie
...
 		}
-
 		ghc := getter.GetGithubClient(token.AccessToken, false)
 		user, err := ghc.GetUser("")
 		if err != nil {
 			ga.serverError(w, "Get user login", err)
 			return
 		}
-		http.SetCookie(w, &http.Cookie{Name: "github_login", Value: *user.Login})
+		http.SetCookie(w, &http.Cookie{
```
Comment (BenTheElder) [inline prow/githuboauth/githuboauth.go]: please go ahead and add `Secure:   true`

File: prow/prstatus/prstatus.go [hunk 2]
```diff
@@ -179,7 +180,11 @@ func (da *DashboardAgent) HandlePrStatus(queryHandler PullRequestQueryHandler) h
 			}
 			// Saves login. We save the login under 2 cookies. One for the use of client to render the
 			// data and one encoded for server to verify the identity of the authenticated user.
-			http.SetCookie(w, &http.Cookie{Name: "github_login", Value: login})
+			http.SetCookie(w, &http.Cookie{
+				Name:    "github_login",
```
Comment (BenTheElder) [inline prow/prstatus/prstatus.go]: and here

**Comment by BenTheElder** [comment]
also maybe consider doing 
```bool
Secure: true,
HttpOnly: true,
```
for the token now, since that's a pretty small and related change

File: prow/githuboauth/githuboauth.go (hunk 1 above)
Comment (qhuynh96) [reply prow/githuboauth/githuboauth.go]: done

File: prow/prstatus/prstatus.go (hunk 2 above)
Comment (qhuynh96) [reply prow/prstatus/prstatus.go]: done

**Comment by stevekuznetsov** [comment]
Changes here

/lgtm
/hold cancel

File: prow/githuboauth/githuboauth.go [hunk 3]
```diff
@@ -160,14 +165,19 @@ func (ga *GithubOAuthAgent) HandleRedirect(client OAuthClient, getter GithubClie
...
+			Name:    "github_login",
+			Value:   *user.Login,
+			Path:    "/",
+			Expires: time.Now().Add(time.Hour * 24 * 30),
```
Comment (BenTheElder) [inline prow/githuboauth/githuboauth.go]: this probably ought to be configurable in the future. we don't have any super major reasons to expire logout, though we may want to add an explicit logout.

**Pull request merged**